### PR TITLE
LibWeb: Support content-visibility css + resolve fixme

### PR DESCRIPTION
**Pull request by EdwinHoksberg**

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
I'll try and look at this later today. I think I spotted a couple of small issues but I've only had time to look through it on my phone. 😅

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWeb/CSS/Properties.json [hunk 1]
```diff
@@ -777,6 +777,17 @@
...
+  "content-visibility": {
+    "animation-type": "none",
+    "__comment": "FIXME: Implement animation https://drafts.csswg.org/css-contain/#content-visibility-animation",
+    "inherited": false,
+    "initial": "visible",
+    "valid-identifiers": [
+      "visible",
+      "auto",
+      "hidden"
+    ]
```
[the remark this answers is not in the document]
Comment (EdwinHoksberg) [reply Userland/Libraries/LibWeb/CSS/Properties.json]: Neat ✨

File: Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp [hunk 2]
```diff
@@ -338,7 +338,7 @@ void TreeBuilder::create_layout_tree(DOM::Node& dom_node, TreeBuilder::Context&
         VERIFY(!element.needs_style_update());
         style = element.computed_css_values();
         display = style->display();
-        if (display.is_none())
+        if (display.is_none() || element.has_parent_with_content_visibility_hidden())
```
[the remark this answers is not in the document]
Comment (EdwinHoksberg) [reply Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp]: That is a lot better yes, I was actually looking for something like that but I totally overlooked that part 😅 Let me know if I implemented it correctly now. 
And yes, pseudo elements should also be invisible, I've added the check there too.

File: Tests/LibWeb/Layout/input/css-content-visibility.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<!doctype html><style>
+div {
+    padding: 16px;
+    content-visibility: hidden;
+}
+</style><div><span>I am not visible
```
[the remark this answers is not in the document]
Comment (EdwinHoksberg) [reply Tests/LibWeb/Layout/input/css-content-visibility.html]: Yes, good idea!

**Review by AtkinsSJ — APPROVED** [review]
Looks good, sorry for the delay! I've had a busy couple of weeks. :sweat_smile: 

There's a conflict that needs fixing, but otherwise this should be good to go.

**Review by AtkinsSJ — APPROVED** [review]
Let's gooooo!

**Pull request merged**